### PR TITLE
driver/screen: support X11 16 bit color frame buffer

### LINKED ISSUE #120: Support more pixel format in x11 screen capture driver
Current code assumes that the pixel format is 32bit RGBA.

### PR DESCRIPTION
**Pull request by at-wat**
### Reference issue
Fixes #120

### CODE REVIEW

File: pkg/driver/screen/x11capture_linux.go [hunk 1]
```diff
@@ -136,6 +198,17 @@ func newShmImage(dp *C.Display, screen int) (*shmImage, error) {
...
 	s := &shmImage{dp: dp}
 
+	switch {
+	case v.red_mask == 0xFF0000 && v.green_mask == 0xFF00 && v.blue_mask == 0xFF:
+		s.pixFmt = pixFmtBGR24
+	case v.red_mask == 0xF800 && v.green_mask == 0x7E0 && v.blue_mask == 0x1F:
+		s.pixFmt = pixFmtBGR16
+	default:
+		fmt.Printf("x11capture: unsupported pixel format (R: %0x, G: %0x, B: %0x)\n",
+			v.red_mask, v.green_mask, v.blue_mask)
```
Comment (at-wat) [inline pkg/driver/screen/x11capture_linux.go]: Error returned from here is silently discarded by `mediaDevices.selectVideo()`. To notify why no driver was selected, output information to stdout.

**Comment by at-wat** [comment]
It can be tested using virtual X11 screen.

```shell
$ Xvfb :10 -ac -screen 0 800x600x16
$ xhost +
```
```shell
$ DISPLAY=:10 go run .
```

**Comment by at-wat** [comment]
`pixFmtBGR24` is completely same as what I named RGBA.
Actual display frame buffer never has alpha value and just a padded to align 64 bits.

**Comment by lherman-cs** [comment]
@at-wat Ah I see, my bad ^^. This looks good to me then!

**Pull request merged**